modules/core/src/main/scala/org/scalasteward/core/forge/gitlab/Url.scala

```diff
@@ -39,9 +39,17 @@ class Url(apiHost: Uri) {
   def existingMergeRequest(repo: Repo, number: PullRequestNumber): Uri =
     mergeRequest(repo) / number.toString
 
-  def mergeWhenPiplineSucceeds(repo: Repo, number: PullRequestNumber): Uri =
+  def mergeWhenPiplineSucceeds(
+      repo: Repo,
+      number: PullRequestNumber,
+      removeSourceBranch: Boolean
+  ): Uri =
     (existingMergeRequest(repo, number) / "merge")
       .withQueryParam("merge_when_pipeline_succeeds", "true")
+      .withOptionQueryParam(
+        "should_remove_source_branch",
+        Option.when(removeSourceBranch)(removeSourceBranch)
```

Comment:
I also found this `should_remove_source_branch` can also be provided as part of `/merge` request so I added as well. When it's `false` query parameter won't be added.

Reply:
> Is the field in MergeRequestPayload not enough for the source branch to be removed after merge?

In theory, it should be but I haven't tested the real behavior to be honest. I found `should_remove_source_branch` last night and I believe it's just another way to remove source branch once merge request is merged.

I think I was too eager to add it but I agree it can be confusing. At the end of the they, if they are doing the same thing we should only have one of them.

I will verify the API behavior and let you know! 

Thanks for taking a look and sorry for the confusion!

Reply:
@fthomas I just reverted the commit where I added the `should_remove_source_branch` query parameter to  "/merge" request.

I tested that adding `remove_source_branch` to `MergeRequestPayload` is sufficient and there is no need for anything additional 👍